### PR TITLE
Openweather api

### PR DESCRIPTION
**Pull request by suzkiee**
# Description
This adds basic service function for MapQuest and OpenWeatherAPI. I set up the MapQuest API since it was paired with the OpenWeatherAPI in SweaterWeather but if it happens that we can get the location coordinates from the Google OAuth instead, we can always ditch it! Just wanted to put it in there to ensure that we can use the OpenWeatherAPI in the upcoming few days. 

This still needs a facade and potential poros but I'm waiting on a convo with frontend to see what information we actually want returned. 

This also adds some missing gems like `faraday`, `json`, and `jsonapi-serializer.` `fast-json` is no longer being maintained so we're actually directed to user the `jsonapi-serializer` gem instead.

This PR also updates the some failing validation tests in the models. 

#Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Closes # (issue)

# How Has This Been Tested?
- [ ] This has not been tested
- [x] This has been tested

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -36,6 +36,11 @@ gem 'jbuilder', '~> 2.5'
 # Reduces boot times through caching; required in config/boot.rb
 gem 'bootsnap', '>= 1.1.0', require: false
 
+# gems for api consumption
+gem 'faraday'
+gem 'jsonapi-serializer'
```
Comment (ztrokey) [inline Gemfile:41]: Good call adding the serializer!

File: app/services/map_quest_service.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
+class MapQuestService
```
Comment (ztrokey) [inline app/services/map_quest_service.rb:1]: This looks great!

**Review by ztrokey — APPROVED** [review]
This all looks awesome! Way to make quick work of it!

File: Gemfile [hunk 3]
```diff
@@ -36,6 +36,11 @@ gem 'jbuilder', '~> 2.5'
...
+gem 'json' 
+
 group :development, :test do
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
```
Comment (denverdevelopments) [inline Gemfile:45]: Did we decide on Travis CI afterall, should we add gem 'travis' ?

**Review by denverdevelopments — APPROVED** [review]

File: Gemfile (hunk 3 above)
Comment (ztrokey) [reply Gemfile:45]: I thought we did but we can clarify in the stand-up today.

**Pull request merged**